airflow/providers/amazon/aws/triggers/redshift_cluster.py

```
@@ -0,0 +1,51 @@
+from typing import AsyncIterator, Any, Tuple, Dict
+
+from airflow.providers.amazon.aws.hooks.redshift_cluster import RedshiftHookAsync
+from airflow.triggers.base import BaseTrigger, TriggerEvent
+
+
+class RedshiftClusterTrigger(BaseTrigger):
+    def __init__(
+        self,
+        task_id: str,
+        aws_conn_id: str,
+        cluster_identifier: str,
+        operation_type: str,
+        poll_interval: float = 5.0,
+    ):
+        super().__init__()
+        self.task_id = task_id
+        self.poll_interval = poll_interval
+        self.aws_conn_id = aws_conn_id
+        self.cluster_identifier = cluster_identifier
+        self.operation_type = operation_type
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        return (
+            "airflow.providers.amazon.aws.triggers.redshift_cluster.RedshiftClusterTrigger",
+            {
+                "task_id": self.task_id,
+                "poll_interval": self.poll_interval,
+                "aws_conn_id": self.aws_conn_id,
+                "cluster_identifier": self.cluster_identifier,
+                "operation_type": self.operation_type,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:
+        hook = RedshiftHookAsync(aws_conn_id=self.aws_conn_id)
+        try:
+            if self.operation_type == "pause_cluster":
```

Should we throw an exception if `operation_type` is not "pause_cluster"